packages/SwingSet/misc-tools/vat-map-from-swingstore.js

```diff
@@ -0,0 +1,155 @@
+// @ts-nocheck
+import '@endo/init';
+import sqlite3 from 'better-sqlite3';
+import fs from 'fs';
+import process from 'process';
+
+// run this like:
+//  node vat-map-from-swingstore.js run-04-swingstore-pruned.sqlite vat-map-04.json 04
+//
+// It will extract data about all vats from that SwingStore DB
+// snapshot, and write it into the .json file. These can be compared
+// with the neighboring vat-map-delta.js . The "04" runID is added to
+// the data for the benefit of quick jq scripts that forget the
+// filename they're looking at.
+
+const [swingstoreDBfn, outputFn, runID] = process.argv.slice(2);
+const ssdb = sqlite3(swingstoreDBfn);
+
+const sqlGet = ssdb
+  .prepare('SELECT value FROM kvStore WHERE key = ?')
+  .pluck(true);
+const get = key => sqlGet.get(key);
+const getJSON = key => JSON.parse(get(key));
+const sqlGetRange = ssdb.prepare(
+  'SELECT * FROM kvStore WHERE key > ? AND key < ?',
+);
+const sqlGetSpan = ssdb.prepare(
+  'SELECT * FROM transcriptSpans WHERE vatID = ? AND isCurrent=1',
+);
+
+const idNumber = (prefix, idString) => {
+  assert(idString.startsWith(prefix));
+  return Number(idString.slice(prefix.length));
+};
+const vatNumber = vatID => idNumber('v', vatID); // 'v12' -> 12
+const sortWith = (arr, keyFunc) => arr.sort((a, b) => keyFunc(a) - keyFunc(b));
```

Comment:
`keyFunc` seems like too much indirection.
```suggestion
const sortNumeric = (arr, prefix) => arr.sort((a, b) => idNumber(prefix, a) - idNumber(prefix, b));
```
Another possibility is just being fully generic, e.g.
```suggestion
const natSort = arr => arr.sort((a, b) => +a.replace(/^\D+/, "") - +b.replace(/^\D+/, ""))
```

Reply:
Huh, I needed to sort both vatIDs and krefs (specifically kdids: kernel-side device IDs), so the second client of `sortWith` is using a different `keyFunc`, one named `kdNumber` (which needs to strip a different prefix, with a different length). Except, I didn't write `kdNumber` properly, it's written as if the argument is a vatID. (I think I wrote `kdNumber` first, then realized a better way to generalize it for vatIDs, but then didn't apply the generalization.

I think I'll keep `keyFunc` but I'll fix `kdNumber` to use it properly.